src/backend/cpu/copy.cpp

```diff
@@ -63,6 +63,30 @@ namespace cpu
         return out;
     }
 
+    template<typename T>
+    void multiply_inplace(Array<T> &in, double val)
```

Comment:
CUDA and OpenCL just wrap other functions. I wrote CPU backend first and did not think about using existing functions. I will change this.